.circleci/config.yml

```diff
@@ -291,6 +298,30 @@ commands:
           name: Wait for local UI server to start
           command: dockerize -wait tcp://localhost:4200 -timeout 4m
 
+  e2e_skip_commit:
+    description: "Halt job if commit message contains 'SKIP E2E'"
+    steps:
+      - run:
+          name: "Halt job if commit message contains 'SKIP E2E'"
+          command: |
+            echo 'export COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")' >> $BASH_ENV
+            source $BASH_ENV
+            echo "Git commit message: $COMMIT_MESSAGE"
+            # Double comma is called "Parameter Expansion" in Bash version 4+. It converts string to lowercase letters.
+            if [[ "${COMMIT_MESSAGE,,}" == *"skip e2e"* ]]; then
```

Comment:
FWIW, it might be cleaner to have  a little Ruby or Python  script to parse the commit message &  other metadata  and give a cleaner structure. (You could even  write test cases for it). My tolerance for reading shell scripts is about 2-1/2 lines.